File: pe-core/src/main/java/org/saar/core/util/Fps.java
package org.saar.core.util;

import org.lwjgl.glfw.GLFW;

public class Fps {

    private double last;

    public Fps() {
        this.last = current();
    }

    private static double current() {
        return GLFW.glfwGetTime();
    }

    public void update() {
        this.last = current();
    }

    public double delta() {
        return current() - this.last;
    }

    public double fps() {
        return 1 / delta();
    }
}

File: pe-core/src/main/java/org/saar/core/util/Fps.kt
package org.saar.core.util

import org.lwjgl.glfw.GLFW

private fun current(): Double {
    return GLFW.glfwGetTime()
}

class Fps {
    private var last: Double = current()

    fun update() {
        this.last = current()
    }

    fun delta() = current() - this.last

    fun fps() = 1 / delta()
}
